server/metadata/tenant.go

```diff
@@ -1051,6 +1051,33 @@ func (tenant *Tenant) ListProjects(_ context.Context) []string {
 	return projects
 }
 
+// DeleteTenant is used to delete tenant and all the content within it.
+// This needs to be followed up with a "restart" of server to clear memory state.
+// Be careful calling this.
+func (m *TenantManager) DeleteTenant(ctx context.Context, tx transaction.Tx, tenantToDelete *Tenant) error {
+	tenantToDelete.RLock()
+	defer tenantToDelete.RUnlock()
+	projects := tenantToDelete.ListProjects(ctx)
+
+	deletedProjects := true
+	for _, project := range projects {
+		_, err := tenantToDelete.DeleteProject(ctx, tx, project)
+		if err != nil {
+			log.Error().Err(err).Msgf("Failed to delete project named %s", project)
+			deletedProjects = false
+			err = tx.Rollback(ctx)
+			if err != nil {
+				log.Error().Err(err).Msgf("Failed to rollback tx while deleting project named %s", project)
```

Comment:
What is the advantage of `deletedProjects` flag? Should we rollback and return?

Reply:
it was carried over when this code was placed outside of tenant manager. fixed thanks.